### PR TITLE
E3035 New rule for DeletionPolicy values

### PR DESCRIPTION
**Pull request by cmmeyer**
*Issue #, if available:*

*Description of changes:*

Add a new rule to validate DeletionPolicy values (Delete,Retain,Snapshot) and make sure Snapshot is used only for allowed resources.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by cmmeyer** [comment]
The code for UpdateReplacePolicy would be almost identical -- I'm debating whether it makes more sense to have two separate rules or evaluate them together?

**Comment by cmmeyer** [comment]
Removing WIP -- this is ready for review.

File: src/cfnlint/rules/resources/DeletionPolicy.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+from cfnlint import RuleMatch
+
+
+class DeletionPolicy(CloudFormationLintRule):
+    """Check Base Resource Configuration"""
+    id = 'E3035'
+    shortdesc = 'Check DeletionPolicy values for Resources'
+    description = 'Check that the DeletionPolicy values are valid'
+    source_url = 'https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html'
+    tags = ['resources', 'dependson']
```
Comment (kddejong) [inline src/cfnlint/rules/resources/DeletionPolicy.py]: not that we use tags much (yet) is dependson correct here?

Comment (cmmeyer) [reply src/cfnlint/rules/resources/DeletionPolicy.py]: Oh shoot. That's a holdover from the rule I stole.. er.. borrowed as a model. I'll clean that up.

**Pull request merged**